homeassistant/components/gotify/__init__.py

```diff
@@ -0,0 +1,30 @@
+"""The gotify integration."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import discovery
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up gotify from a config entry."""
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = entry
+    # Gotify notify component requires entry_id in discovery_info to get the correct entry config.
+    # The discovery platform requires CONF_NAME in discovery_info to set a service name.
+    discovery_info = {CONF_NAME: entry.data[CONF_NAME], "entry_id": entry.entry_id}
+    await hass.async_create_task(
```

Comment:
I think the typical pattern for this has changed to:

```python
PLATFORMS = [Platform.NOTIFY]  # Somewhere above in the module

hass.config_entries.async_setup_platforms(entry, PLATFORMS)
```

I'm not sure if that warrants changing or not.
```

Reply:
The notify platform doesn't support config entries yet. You have to use discovery to load platform.

Reply:
Ah, good to know!